### PR TITLE
Linear regression

### PR DESCRIPTION
**Pull request by MarHai**
Implements the `regress` function as a wrapper to `lm`. 

Accepts only factor and numeric variables as predictors and only one numeric variable as outcome. Checks and stops if violated. If input data is grouped, groups are dropped and a warning is issued.

Three optional checks are included (all added through messages on top of the returned tibble):

- independence of errors (through a Durbin-Watson test from the `car` package)
- multicollinearity (through VIF and tolerance from the `car` package); only feasible if at least two predictors are given and if none of them is a factor (ignores checks and warns about these preconditions if violated)
- homoscedasticity (through a Breusch-Pagan test from the `car` package)

Returns a tibble and, by default, includes beta values (from the `lm.beta` package). Informs (through messages on top of its returned tibble) about goodness of fit and, if applicable, about the reference level used for factor predictors.  

Currently missing is the additional model as an attribute. This is added once @joon-e has added that to the rest of the package.

### CODE REVIEW

File: R/tidycomm-package.R [hunk 1]
```diff
@@ -1,8 +1,11 @@
...
-#' @importFrom utils combn hasName
-## usethis namespace: end
-NULL
+## usethis namespace: start
+#' @import rlang
+#' @importFrom car durbinWatsonTest ncvTest vif
+#' @importFrom glue glue
+#' @importFrom lm.beta lm.beta
+#' @importFrom stats chisq.test cor cor.test median na.omit quantile sd t.test var TukeyHSD aov as.formula
+#' @importFrom stats pf
```
Comment (LKobilke) [inline R/tidycomm-package.R:7]: add "pf" to "stats" import in line 6

**Review by LKobilke — APPROVED** [review]
No errors have arisen. Feedback contains additional ideas and details that do not create errors. Regardless of addressing, merging can be approved. I am still new to testthat(), so I cannot yet assess here if further tests are needed.

File: R/regression.R [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+#' WoJ %>% regress(autonomy_selection, ethics_1)
+#' WoJ %>% regress(autonomy_selection, work_experience, trust_government)
+#'
+#' @export
+regress <- function(data,
+                    dependent_var,
+                    ...,
+                    check_independenterrors = FALSE,
+                    check_multicollinearity = FALSE,
+                    check_homoscedasticity = FALSE) {
```
[the remark this answers is not in the document]
Comment (MarHai) [reply R/regression.R:35]: If the regression contains factors then a warning by `check_multicollinearity` will be issued. My thought was to (a) do not have the default settings raise warnings for a valid regression model and (b) be coherent across all three parameters. 

Put the other way round: Why would anybody set it to `FALSE` if they default to `TRUE`? If we want to present them as optional, to me, having them default to `FALSE` seems more natural.

That said, I am also happy to change them to `TRUE`.

File: R/regression.R [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+                    check_durbin_watson$dw,
+                    check_durbin_watson$p))
+  }
+
+  if (check_homoscedasticity) {
+    check_breusch_pagan <- car::ncvTest(model)
+    message(sprintf("- Check for homoscedasticity: Breusch-Pagan = %f (p = %f)",
+                    check_breusch_pagan$ChiSquare,
+                    check_breusch_pagan$p))
+  }
```
[the remark this answers is not in the document]
Comment (MarHai) [reply R/regression.R:126]: If we want to integrate visual inspection then why only do that for regression? 

Also, once we change the return types in all functions from `tibble` to a tidycomm class (with secondary `tibble`), we could also integrate `plot` functions separately that can respond to the correct type. The code would then look something like this:

```r
WoJ %>%
  regress(autonomy_selection, work_experience, trust_government) %>%
  visualize()

WoJ %>%
  t_test(work_experience, trust_government) %>%
  visualize()
```

File: R/regression.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (LKobilke) [reply R/regression.R:35]: I see. Then it makes sense to keep `FALSE` as the default, in order to reduce the amount of raised warnings. In addition, students then need to reflect on whether a multicollinearity check is applicable, which I had overlooked.

File: R/regression.R (hunk 3 above)
[the remark this answers is not in the document]
Comment (LKobilke) [reply R/regression.R:126]: Absolutely **love** this idea! Very seamless!

File: R/regression.R [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+#'   fitted. Specify as column name.
+#' @param ... Independent variables to take into account as (one or many)
+#'   predictors for the dependent variable. Specify as column names. At least
+#'   one has to be specified.
+#' @param check_independenterrors if set, the independence of errors among any
+#'   two cases is being checked using a Durbin-Watson test
+#' @param check_multicollinearity if set, multicollinearity among all specified
+#'   independent variables is being checked using the variance inflation factor
+#'   (VIF) and the tolerance (1/VIF); can only be calculated if at least two
+#'   independent variables are given all of which are numeric
```
[the remark this answers is not in the document]
Comment (MarHai) [reply R/regression.R:19]: #ee48c6c

File: R/regression.R (hunk 3 above)
[the remark this answers is not in the document]
Comment (MarHai) [reply R/regression.R:126]: #23

File: R/tidycomm-package.R (hunk 1 above)
Comment (MarHai) [reply R/tidycomm-package.R:7]: #c172c3e

**Review by LKobilke — APPROVED** [review]

**Pull request merged**